src/rgw/driver/rados/rgw_notify.h

```diff
@@ -117,5 +117,8 @@ int publish_commit(rgw::sal::Object* obj,
 // cancel the reservation
 int publish_abort(reservation_t& reservation);
 
+int get_persistent_queue_by_topic_name(const DoutPrefixProvider* dpp, librados::IoCtx& rados_ioctx, const std::string& topic_name,
+                                       std::size_t& number_of_reservations, uint32_t& number_of_entries, uint64_t& queue_size, optional_yield y);
```

Comment:
maybe better to create a "stats" struct and pass it.
later it could be used to pass this data in other places.
also, you can add a JSON formatter to the struct

Reply:
Ack!